src/cpp/redisserver.cpp

```diff
@@ -90,3 +111,79 @@ void RedisServer::_check_ssdb_string(const std::string& env_str) {
         }
     }
 }
+
+//Initialize variable of type integer from environment variable
+void RedisServer::_init_integer_from_env(int& value,
+                                         const std::string& env_var,
+                                         const int& default_value)
+{
+    value = default_value;
+
+    char* env_char = getenv(env_var.c_str());
+
+    if (env_char != NULL && strlen(env_char) > 0) {
+        // Enforce that all characters are digits because std::stoi
+        // will truncate a string like "10xy" to 10.
+        // We want to guard users from input errors they might have.
+        char* c = env_char;
+        while (*c != 0) {
+            if (!isdigit(*c) && !(*c == '-' && c == env_char)) {
+                throw SRRuntimeException("The value of " + env_var +
+                                         " must be a valid number.");
+            }
+            c++;
+        }
+
+        try {
+            value = std::stoi(env_char);
+        }
+        catch (std::invalid_argument& e) {
+            throw SRRuntimeException("The value of " + env_var + " could "\
+                                     "not be converted to type integer.");
+        }
+        catch (std::out_of_range& e) {
+            throw SRRuntimeException("The value of " + env_var + " is too "\
+                                     "large to be stored as an integer "\
+                                     "value.");
+        }
+        catch (std::exception& e) {
+            throw SRRuntimeException(e.what());
```

Comment:
Add some text to give context or this is going to be really mysterious if we ever catch something here.
Also, make it an SRInternalError -- should never catch one of these